Source/DynamicTradeInterface/InterfaceComponents/TableBox/Table.cs

```diff
@@ -349,15 +345,14 @@ public void Draw(Rect boundingBox)
 			float tableWidth = boundingBox.width - GenUI.ScrollBarWidth - CELL_SPACING;
 			float leftoverWidth = tableWidth - _fixedColumnWidth - (CELL_SPACING * _columns.Count);
 
-			TableColumn<T> column;
 			if (_drawHeaders)
 			{
 				Rect columnHeader = new Rect(boundingBox);
 				columnHeader.height = Text.LineHeight;
 
 				for (int i = 0; i < _columns.Count; i++)
 				{
-					column = _columns[i];
+					TableColumn<T> column = _columns[i];
```

Comment:
In C++ it is better to define stack variables inside of for-loops when its value must not be retained between iterations. Since this restricts the scope of the variable, the for-loop is simpler and the compiler can reason better about it, leading to better optimizations in most cases. The variable allocation itself happens in the stack in both cases, so in that particular point both approaches are the same.

In C#, Rider's suggested this mostly to reduce the scope of the variable and avoid potential errors. I followed it due mostly to this reason, but it is a good idea to check what happens in the underlying IL code. Here is a short example using SharpLab that I have made to test this case.

https://sharplab.io/#v2:C4LglgNgPgAgTARgLACgYAYAEMEBYDcqqMAzNnJgMKYDeqmD2ZMumA8gK7ADOYAJgFMAFAEpa9RpIASAQ24ALAMoDgAHhzoAfJgC2AT2XBMAXkwA7AQHdMshYfUItowikmSNmAE4DuHCMBc3RgAzAHtPTCEwMyMwE0x0fEw41UxHRMwAakywEQkgulcgt29ff3iAEgAiADUZCA4BEFowAF8qwOLJfUMAOgBBPj4hUr9gEU6g1vy3HABOIR6VXspQjhiJmemihhnSbFYASTNeQVFxHbdbJRUHLV0DFXiLa2v7DU1nGckwiKiY5LxDIpNLoDLZXLfRiFLruRxeHxjSq1eqNZo0NodKFBJbAAZDEaI/ybS6SbZdeaLR541brcaTTDbVpAA=

Judging from this example, C# has a similar situation than C++. For both Outside and Inside, `string result` is being defined in the stack, using `.locals init`. The generated IL instructions are the same for both methods, except for the locals array having variables on different indexes. The same situation happens in both Release and Debug builds. StackOverflow answers about this topic confirm that this holds true in other similar cases as well.

Since the C# compiler is smart enough to perform these optimizations when the variable is declared inside of the for-loop, I think it is better to follow this approach as it has the added benefit of preventing using the variable in unintended ways.